tests/ping.go

```diff
@@ -98,3 +98,38 @@ func vmiConsoleExpectBatch(vmi *v1.VirtualMachineInstance, expected []expect.Bat
 	}
 	return err
 }
+
+// PingAppJob performs a netcat check (tcp by default) using a pod.
+// Optional port and arguments for the netcat command may be provided.
+// If port is not provided, port "8080" is used.
+func PingAppJob(host, port string, args ...string) error {
+	const (
+		netcat      = "nc"
+		defaultPort = "8080"
+
+		jobRetry   = 1
+		jobTTL     = 10
```

Comment:
Any reason to use only 10? You don't assert here so the caller of this function can assert after... I am just worrying to lose some valuable debug info...

Reply:
10 seconds is a long of time and our usage of jobs in its current state is anyway problematic because a large number of them will be left behind, uncleaned. (we need to take care of that and add it to the namespace cleanup function and probably provide a delete API for jobs).

This helper will create the the job, then wait for it to finish and return. At the same time (more or less), the 10sec timer will start so the caller has now ~10sec to check something. If he decides to assert immediately based on the returned result, there is no problem.

If he will check something else and based on that result assert, then we may get into a problem of not collecting the job logs as you mentioned. In such a case, no number we will use here will really answer the issue because it depends on how long that other check will take.
One generic solution I can think of is to require a timeout value to be passed in to this helper, but I feel this will not make sense to the caller and will complicate the helper signature.

What I think we can do (in a follow up PR hopefully) is to provide a helper that collects the logs from job pods, letting the caller explicitly fetch the information in the scenario mentioned above.

In the meantime, I will increase this to 15 (as the caller is indeed checking another operation for 10sec max).
I will also document this on the doctext.

Reply:
I just realized that this feature is still feature-gated. (Sorry, I am using it in my project...) I assume we don't have this enabled so indeed we need to have our own clean-ups so we don't dirty other tests. (follow up PR is good)

Reply:
If this is the case (that it is feature gated), we have a problem. The TTL cannot be an API of the job abstraction nor for this ping one. An alternative is to launch a goroutine to do this instead, but I do not know if it is worth the trouble as other objects do not do the same at the moment.

Reply:
I think it would be best to add the clean-up of jobs after each test.